src/synthesized_insight/check.py

```diff
@@ -6,9 +6,50 @@
 
 
 class Check(ABC):
-    """Abstract class for checking column types
-    Implement this class for the customized logic for different types of checks
+    """Class for checking column types
+    Implement the methods of this class for the customized logic for different types of checks
     """
+
+    def infer_dtype(self, sr: pd.Series) -> pd.Series:
```

Comment:
The method infer_dtype is moved to base class Check because the column type inference is closely related to column checks (i,e. it happens only when the column type needs to be checked). Also, the infer_dtype method is used in modeling preprocessor class which won't be customized in the SDK, and we will use the SDK's customized version of Check class to initialize the preprocessor class object in SDK, hence we need access to this method through the SDK's customized version of Check class as well.